File: app/src/main/java/com/pengxh/ncnn/yolov8/INativeCallback.java
package com.pengxh.ncnn.yolov8;

import java.util.ArrayList;

public interface INativeCallback {
    /**
     * 分类
     */
    void onClassify(float[] possibles);

    /**
     * 分割
     */
    void onPartition(ArrayList<String> output);

    /**
     * 检测
     */
    void onDetect(ArrayList<String> output);
}

File: app/src/main/java/com/pengxh/ncnn/yolov8/INativeCallback.kt
package com.pengxh.ncnn.yolov8

interface INativeCallback {
    /**
     * 分类
     */
    fun onClassify(possibles: FloatArray)

    /**
     * 分割
     */
    fun onSegmentation(
        segmentationOutput: ArrayList<FloatArray>, detectOutput: ArrayList<FloatArray>
    )

    /**
     * 检测。只需要返回类别
     */
    fun onDetect(output: ArrayList<String>)
}